appside/frames/likert.js

```diff
@@ -0,0 +1,114 @@
+"use strict";
+
+/**
+ * Rendering (View) code for likert frame
+ * @author Rachel Sitt
+ */
+
+
+/**
+ * A frame composed of a list of things.
+ * 
+ * Things in the list are to be rendered as a list of checkboxes.
+ **/
+class LikertFrame extends Frame {
+
+    /**
+     * Construct LikertFrame from an object
+     * 
+     * @param frame_data -- Object containing the frame's data. Expected fields:
+     *    frame_data.question (string) -- Text to appear before the list of statements
+     *  Behavior undefined if frame does not have these properties.
+     */
+    constructor(frame_data) {
+        super();
+
+        // set fields
+        this.title = LIKERT_TITLE;
+        this.instructions = LIKERT_INSTRUCTIONS;
+        this.questions = frame_data.questions;
+        this.user_input = new Map();
+    }
+
+    /**
+     * Render this frame into the DOM
+     * 
+     * @require -- DOM must have a div whose ID is 'frame'
+     * 
+     * @effects -- Does not preserve former content of <div id="frame">.
+     *    Renders the data from this into that div.
+     */
+    render() {
+
+        // make a new empty div with id frame, not yet in the dom
+        let frame = document.createElement('div'); 
+        $(frame).attr('id', 'frame');
+        
+        // insert a h5 node for the title
+        let title = document.createElement('h5');
+        $(title).text(this.title);
+        $(title).attr('class', 'text-info text-uppercase mb-2');
+        frame.appendChild(title);
+        
+        // insert a h5 node for the instruction
+        let instructions = document.createElement('h5');
+        $(instructions).text(this.instructions);
+        frame.appendChild(instructions);
+
+        // insert a radio button list for the statements
+        let statements = document.createElement('div');
+        $(statements).attr('class', 'form-check');
+
+        let i = 0;
+        for (let data of this.questions) {
+            let question = data[0];
+            let answer = data[1];
+            i += 1;
+
+            let question_text = document.createElement('h5');
+            $(question_text).attr('class', 'likert_question_text');
+            $(question_text).text(data);
+            statements.appendChild(question_text);
+            this.user_input.set(question, undefined);
+
+            // the actual radio buttons
+            for (let j = 1; j <= 5; j++) {
+                let input = document.createElement('input');
+                $(input).attr('class', 'form-check-input');
+                $(input).attr('class', 'likert_input');
+                $(input).attr('type', 'radio');
+                $(input).attr('name', question);    // question text
+                input.dataset.text = j;             // answer choice
+
+                let input_text = document.createElement('p');
+                $(input_text).attr('class', 'likert_input_text');
+                $(input_text).text(j);
```

Comment:
try to make the input text be a "label" so that when you click the text, the radio button gets selected

Reply:
Created an event listener on each input_text when clicked to 'check' the corresponding radio button.

Reply:
Debugged and made input_text as labels, input's id as _question + [1-5]_.